File: hashtable/1.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] ret = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int left = target - nums[i];
            if (map.get(left) != null) {
                ret[0] = map.get(left);
                ret[1] = i;
                break;
            }else {
                map.put(nums[i], i);
            }
        }
        return ret;
    }
}

File: hashtable/1.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::vector<int> ret(2, 0);
        std::unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); ++i) {
            auto iter = m.find(target-nums[i]);
            if (iter != m.end()) {
                ret[0] = iter->second;
                ret[1] = i;
                return ret;
            }
            m[nums[i]] = i;
        }
        return ret;
    }
};
